ttnn/cpp/ttnn/operations/normalization/layernorm/device/kernels/compute/layernorm_sharded_post_allgather.cpp

```diff
@@ -0,0 +1,333 @@
+                cb_pop_front(cb_stats, 1);
+            }
+            tile_regs_commit();
+            tile_regs_wait();
+
+            #ifdef RMSNORM
+            pack_tile(dst0, cb_var);
+            #else
+            pack_tile(dst0, cb_stats_reduced);
+            pack_tile(dst1, cb_ex2);
+            #endif
+            tile_regs_release();
+            reduce_revert_delta();
+            #ifdef RMSNORM
+            cb_push_back(cb_var, stats_tiles);
+            #else
+            cb_push_back(cb_stats_reduced, 1);
+            cb_push_back(cb_ex2, 1);
+            #endif
+
+            #ifndef RMSNORM
+            // calculate var = E(x^2) - E(x)^2
+            // E(x)^2
+            unpack_reconfig_data_format(cb_stats_reduced, cb_stats_reduced);
+            cb_reserve_back(cb_ex_sqr, 1);
+            cb_wait_front(cb_stats_reduced, 1);
+            tile_regs_acquire();
+            mul_tiles_init();
+            mul_tiles(cb_stats_reduced, cb_stats_reduced, 0, 0, dst0);  // first tile in stats is always E(x)
+            tile_regs_commit();
+            tile_regs_wait();
+            pack_tile(dst0, cb_ex_sqr);
+            cb_push_back(cb_ex_sqr, 1);
+            tile_regs_release();
+
+
+            // E(x^2) - E(x)^2
+            unpack_reconfig_data_format_srca(cb_stats_reduced, cb_ex2);
+            unpack_reconfig_data_format_srcb(cb_stats_reduced, cb_ex_sqr);
+            pack_reconfig_data_format(cb_var);
+            cb_wait_front(cb_ex2, 1);
+            cb_wait_front(cb_ex_sqr, 1);
+            cb_reserve_back(cb_var, 1);
+            tile_regs_acquire();
+            sub_tiles_init();
+            sub_tiles(cb_ex2, cb_ex_sqr, 0, 0, dst0);
+            tile_regs_commit();
+            tile_regs_wait();
+            pack_tile(dst0, cb_var);
+            cb_push_back(cb_var, 1);
+            tile_regs_release();
+            cb_pop_front(cb_ex2, 1);
+            cb_pop_front(cb_ex_sqr, 1);
+            #endif
+
+
+            // 1/[sqrt(Var + eps)],
+            unpack_reconfig_data_format(cb_var, cb_eps);    // cb_var is cb_stats in case of RMS norm
+            pack_reconfig_data_format(cb_stats_reduced);
+            cb_wait_front(cb_var, 1);
+            cb_wait_front(cb_eps, 1);
+            cb_reserve_back(cb_stats_reduced, 1);
+
+            add_tiles_init();
+            tile_regs_acquire();
+            add_tiles(cb_var, cb_eps, 0, 0, dst0);
+            tile_regs_wait();
+            sqrt_tile_init();
+            sqrt_tile(dst0);
+            recip_tile_init();
+            recip_tile(dst0);
+            tile_regs_commit();
+            tile_regs_wait();
+            pack_tile(dst0, cb_stats_reduced);
+            tile_regs_release();
+            cb_pop_front(cb_var, 1);
+            cb_pop_front(cb_eps, 1);
+            cb_push_back(cb_stats_reduced, 1);
+        }
+    }
+
+    #ifndef RMSNORM
+    // x - E[x]
+    unpack_reconfig_data_format(cb_in0, cb_ex_global);
```

Comment:
Should we use the calls unpack_reconfig_data_format_srca/srcb instead? Since they are not forcing an update?

Reply:
Actually, we need force here as non-all-to-all workers directly jump here and previously srca = cb_var , srcb = cb_eps